src/crypto/algorithms/megolm.js

```diff
@@ -402,6 +400,17 @@ MegolmEncryption.prototype._splitUserDeviceMap = function(
 
             entriesInCurrentSlice++;
         }
+
+        // We do this in the per-user loop as we prefer that all messages to the
+        // same user end up in the same API call to make it easier for the
+        // server (e.g. only have to send one EDU if a remote user, etc). This
+        // does mean that if a user has many devices we may go over the desired
+        // limit, but its not a hard limit so that is fine.
```

Comment:
Going over the limit is probably also fine now that the payload size is much smaller since we don't include everyone's ciphertext any more.